### PR TITLE
Update of TC plugin

### PR DESCRIPTION
**Pull request by Syndaryl**
- Refactored material stat generation to use a factory method
- Set up a draft of an automated conversion from BM/MM statistics to TC statistics. This way if anything changes, the TC stats automatically update. 
- Should now be fairly easy to port to other BM-based mods that add new metals.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -23,4 +23,6 @@ run
 
 /README.txt
 /forge-*-changelog.txt
-gradle.properties
\ No newline at end of file
+gradle.properties
+libs/Tinkers/Tinkers-src.zip
```
Comment (iLexiconn) [inline .gitignore:27]: ?

File: src/main/java/modernmetals/init/MaterialCorrelation.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+import cyano.basemetals.material.MetalMaterial;
+import net.minecraftforge.fluids.Fluid;
+import slimeknights.tconstruct.library.materials.Material;
+
+class MaterialCorrelation {
+	@SuppressWarnings("unused")
+	private Material material;
+	@SuppressWarnings("unused")
+	private MetalMaterial metal;
+	@SuppressWarnings("unused")
```
Comment (iLexiconn) [inline src/main/java/modernmetals/init/MaterialCorrelation.java]: not needed

**Review by iLexiconn — CHANGES_REQUESTED** [review]
Also fix some of the formatting please

File: .gitignore (hunk 1 above)
Comment (Syndaryl) [reply .gitignore:27]: The .gitignore shouldn't be in the PR.

File: src/main/java/modernmetals/init/MaterialCorrelation.java (hunk 2 above)
Comment (Syndaryl) [reply src/main/java/modernmetals/init/MaterialCorrelation.java]: Corrected.

**Review by iLexiconn — APPROVED** [review]

**Pull request merged**